code/modules/power/engines/tesla/energy_ball.dm

```diff
@@ -33,6 +34,18 @@
 	var/energy_to_lower = -20
 	var/list/shocked_things = list()
 	var/obj/singularity/energy_ball/parent_energy_ball
+	/// Variable to determine if it has a target
+	var/has_a_target = FALSE
```

Comment:
cant you just do `if(target_turf)` then whenever you would have set this variable to false set `target_turf` to `null`

Reply:
That's me having a brainfart honestly, will update it